app/src/main/java/com/dragic/gamehunter/viewmodel/HomeViewModel.kt

```diff
@@ -19,12 +19,35 @@ class HomeViewModel @Inject constructor(
     var dealData by mutableStateOf<List<DealViewState>>(emptyList())
         private set
 
+    private var pageNumber = 1
+
+    private var shouldIncrement = true
+
     init {
         viewModelScope.launch {
-            dealData = repository.dealData().map { it.toDealViewState() }
+            dealData = repository.dealData(pageNumber).map { it.toDealViewState() }
+        }
+    }
+
+    fun loadNextPage() {
+        viewModelScope.launch {
+            dealData = buildList {
+                addAll(dealData)
+                addAll(repository.dealData(pageNumber).map { it.toDealViewState() })
+            }
+            shouldIncrement = true
         }
     }
 
+    fun incrementPageNumber() {
+        if (shouldIncrement) {
+            pageNumber++
+            shouldIncrement = false
+        }
+    }
+
+    fun checkNotEmpty() = dealData.isNotEmpty()
```

Comment:
The ViewModel should never have public methods that return something. It should also never have public variables that are not lifecycle-aware observables like LiveData, MutableState or flows. In other words, this is bad: 

```kt
class BadViewModel : ViewModel() {
   var publicallyMutableVariable = 1
   
   fun isPublicallyMutableVariable2() = publicallyMutableVariable == 2
}
```

Instead, the ViewModel should be set up so that events come in via void methods, and state that's a product of that event comes out in a stream. 

```kt
class GoodViewModel : ViewModel() {
   private var privateVariable = 1
   var is2 by mutableStateOf(privateVariable == 2)
       private set
   
   fun incrementVariable() {
     privateVariable++
     is2 = privateVariable == 2
   }
}
``` 

Having methods that return values and public variables alongside the state variable gets tangled and unmaintainable pretty quickly.